docs/configuration.md

```diff
@@ -0,0 +1,27 @@
+Configuration
+=============
+
+A `config.yaml` file must exist at the root of the repository. See [the schema](../schemas/config.json) to find all possible properties and what they are used for.
+
+Note:
+* The bucket `referenceName` acts as an alias for referencing the bucket. If a `referenceName` is added it must be used to reference the bucket.
+* If you are using Elasticsearch via AWS the region portion of the Elasticsearch config is required and the AWS keys are only required when the Elasticsearch Domain access policy requires keys.
+* `upperSearchResultLimit` is another optional Elasticsearch config option. It defaults to 10000 if not set. It limits the number of search results returned. We currently do not support pagination.
+
```

Comment:
Can you add a note about not being able to have a space in `hookCommand` filename?